test/e2e/lib/components/site-editor-component.js

```diff
@@ -258,33 +258,55 @@ export default class SiteEditorComponent extends AbstractEditorComponent {
 		);
 	}
 
-	async saveGlobalStyles( { pauseAfter = false } = {} ) {
+	async maybeSaveGlobalStyles( { pauseAfter = false } = {} ) {
+		// Bail early if saving is not enabled.
+		const isNothingToSave = await driverHelper.isElementLocated(
```

Comment:
It seems a little odd to me that we're adding this guard into `saveGlobalStyles`.

Why we would be uncertain about whether or not there are global styles updates that should be saved if we're the ones writing the specs?

Is this to patch non-deterministic behavior in the e2e testing environment?

Reply:
Basically.  Previously when we reset the global styles, there would be something to save regardless of whether or not it was really on a clean slate.  Now with recent updates, resetting styles only dirties the new entity if there really is a change.

The reason we have the safeguard is because due to possibly test failures we don't know how dirty global styles will be when we start the test.  If the previous run failed after setting some global styles but before resetting, then we have something to reset at the beginning of the next test run.  If the previous test run went fine, then things were reset appropriately and we dont have anything to reset at the beginning of the nexxt test run. etc.

Reply:
> The reason we have the safeguard is because due to possibly test failures we don't know how dirty global styles will be when we start the test.

Gotcha. If I understand correctly, this safeguard is meant to prevent cascading failures caused by a botched test. It seems like, ideally, we'd have some sort of `afterEach` or `after` callback (akin to Mocha) that could reset global styles when things don't behave as expected.

I think it might be worthwhile to add an inline comment. I'll do some light digging in the meantime, but I don't think this is a blocker.

cc @worldomonation

Reply:
Right, we moved cleanup to be consistently in the `after` calls here.  Calling it early as a precaution is just an extra safety.  

The only time we need to save global styles in e2e is either when we are cleaning up, or testing the save GS tracking event.  In case 1 its ok to conditionally save in the case that they have already been reset, in case 2 if the save doesnt happen the the tracking events arent sent and the tests fail.